websocket_back/src/room.js

```diff
@@ -0,0 +1,80 @@
+export default class Room {
+
+  constructor (options) {
+    this.io = options.io;
+    this.socket = options.socket;
+    this.roomId = options.roomId;
+    this.username = options.username;
+    // join | create
+    this.action = options.action;
+    this.store = this.io.adapter;
+  }
+
+  async init () {
+
+    let clients;
+    await this.io.in(this.roomId).clients((e, _clients) => {
+      clients = _clients || console.error('[INTERNAL ERROR] Room creation failed!');
+      console.log(`Connected Clients are: ${clients}`);
+    });
+
+    if (this.action === 'join') {
+
+      if (clients.length >= 1) {
+        await this.socket.join(this.roomId);
+        this.store.clients.push({ id: this.socket.id, username: this.username, isReady: false });
+        this.socket.username = this.username;
+        this.socket.emit('join_room', '[SUCCESS] Successfully initialised');
+        console.log(`[JOIN] Client joined room ${this.roomId}`);
+        return true;
+      }
+
+      console.warn(`[JOIN FAILED] Client denied join, as roomId ${this.roomId} not created`);
+      this.socket.emit('join_room', 'Error: Create a room first!');
+      return false;
+    }
+
+    if (this.action === 'create') {
+
+      if (clients.length < 1) {
+        await this.socket.join(this.roomId);
+        this.store.clients = [{ id: this.socket.id, username: this.username, isReady: false }];
+        this.socket.username = this.username;
+        console.log(`[CREATE] Client created and joined room ${this.roomId}`);
+        this.socket.emit('create_room', '[SUCCESS] Successfully initialised');
```

Comment:
Same as the join here : 
```suggestion
        this.socket.emit('create_room', '[SUCCESS] Successfully initialised', this.roomId);
```